qir/qat/RemoveDisallowedAttributesPass/RemoveDisallowedAttributesPass.hpp

```diff
@@ -21,20 +21,28 @@ namespace quantum
         RemoveDisallowedAttributesPass()
           : allowed_attrs_{
                 static_cast<String>("EntryPoint"), static_cast<String>("InteropFriendly"),
-                static_cast<String>("requiredQubits"), static_cast<String>("requiredResults")}
+                static_cast<String>("requiredQubits"), static_cast<String>("requiredResults")
```

Comment:
If the `String` here is an alias of `std::string` then what is the advantage of 
`static_cast<String>("requiredQubits")` over the 
`String("requiredQubits")`?

Reply:
As far as I am aware there is no difference as `static_cast<String>("EntryPoint")` should translate into `String("EntryPoint")`. It is only a semantic difference in that we are casting `char const *` to `String`, rather than creating `String`. In the case of`std::string` I believe it is even possible to get away with implicit casts (but I might remember wrong, but I don't think the constructor is marked as explicit).